feat(volumeFee): add volume fee for Safe

shoom3301 opened the pull request:
# Summary

Reference: https://help.safe.global/en/articles/178530-how-does-the-widget-fee-work-for-native-swaps

Basically, CoW Swap should work the same with https://app.safe.global/swap when it is open in Safe Apps.
I literally copy-pasted the logic from Safe frontend.
The only thing that might be different is stablecoins list. We have [our own list](https://github.com/cowprotocol/cowswap/blob/4cf0c91a300d6fcdc026364e74c45e5b2f729f2b/libs/common-const/src/tokens.ts#L488), our list at least matches mainnet (USDC/USDT/DAI).

<img width="582" alt="image" src="https://github.com/user-attachments/assets/4f31aca0-7abe-4230-b03b-7d1bfa205f1b">

# To Test

Volume fee should work exactly the same with https://app.safe.global/swap

anxolin commented on apps/cowswap-frontend/src/modules/volumeFee/state/safeAppFeeAtom.ts:
```diff
@@ -0,0 +1,66 @@
+ */
+export const safeAppFeeAtom = atom<VolumeFee | null>((get) => {
+  const { chainId } = get(walletInfoAtom)
+  const { isSafeApp } = get(walletDetailsAtom)
```
will this affect also wallet connect, or just the safe app?

shoom3301 replied:
Just Safe App

anxolin reviewed (approved):
Looks great! nice job

elena-zh reviewed (changes requested):
Hey @shoom3301 , great, but: 
1. Fee breakdowns are not working. No matter which amount to sell I select. the min fee size is applied to a trade
2. It would be nice to remove intends in this tooltip (we don't have them in other tooltips)
![remove spaces in the tooltip](https://github.com/user-attachments/assets/81c02c95-9b4a-4a39-b11f-fce23d5b534e)
3. It would be great to add a feature flag for this feature
4. [Question](https://cowservices.slack.com/archives/C0361CDG8GP/p1732884356364209): We are loosing our internal fees on Arb and GC. (0.1%). Are we OK with this? 
5. Should we rename this fee inside the AppData ([discussion](https://cowservices.slack.com/archives/C0361CDG8GP/p1732885113822479))?
![image](https://github.com/user-attachments/assets/09e195c1-f488-48a5-a96f-1b983b9e851f)

shoom3301 commented:
@elena-zh 
1. Fixed
2. Fixed
3. https://app.launchdarkly.com/projects/default/flags/isSafeAppFeeEnabled/targeting?env=production&env=staging&env=barn&env=develop&env=test&selected-env=production
4. IDK, let's wait
5. No

elena-zh reviewed (approved):
Thank you for clarfication/fixes. 
The flag is working, fee breakdowns are working. 
2 things to clarify are left; 
1. Should we change fees on Base?
2. make sure that recipient address is correct.

elena-zh commented:
I've done one more test round: everything works great!
